File: tests/testdata/codegen/typed-enum-pp/dotnet/typed-enum.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using AzureNative = Pulumi.AzureNative;

return await Deployment.RunAsync(() => 
{
    var someString = "foobar";

    var typeVar = "Block";

    var staticwebsite = new AzureNative.Storage.StorageAccountStaticWebsite("staticwebsite", new()
    {
        ResourceGroupName = someString,
        AccountName = someString,
    });

    // Safe enum
    var faviconpng = new AzureNative.Storage.Blob("faviconpng", new()
    {
        ResourceGroupName = someString,
        AccountName = someString,
        ContainerName = someString,
        Type = AzureNative.Storage.BlobType.Block,
    });

    // Output umsafe enum
    var _404html = new AzureNative.Storage.Blob("_404html", new()
    {
        ResourceGroupName = someString,
        AccountName = someString,
        ContainerName = someString,
        Type = staticwebsite.IndexDocument.Apply(System.Enum.Parse<AzureNative.Storage.BlobType>),
    });

    // Unsafe enum
    var another = new AzureNative.Storage.Blob("another", new()
    {
        ResourceGroupName = someString,
        AccountName = someString,
        ContainerName = someString,
        Type = System.Enum.Parse<AzureNative.Storage.BlobType>(typeVar),
    });

});

File: tests/testdata/codegen/typed-enum-pp/nodejs/typed-enum.ts
import * as pulumi from "@pulumi/pulumi";
import * as azure_native from "@pulumi/azure-native";

const someString = "foobar";
const typeVar = "Block";
const staticwebsite = new azure_native.storage.StorageAccountStaticWebsite("staticwebsite", {
    resourceGroupName: someString,
    accountName: someString,
});
// Safe enum
const faviconpng = new azure_native.storage.Blob("faviconpng", {
    resourceGroupName: someString,
    accountName: someString,
    containerName: someString,
    type: azure_native.storage.BlobType.Block,
});
// Output umsafe enum
const _404html = new azure_native.storage.Blob("_404html", {
    resourceGroupName: someString,
    accountName: someString,
    containerName: someString,
    type: staticwebsite.indexDocument,
});
// Unsafe enum
const another = new azure_native.storage.Blob("another", {
    resourceGroupName: someString,
    accountName: someString,
    containerName: someString,
    type: azure_native.storage.BlobType[typeVar],
});
